lib/lib-doi/src/main/java/org/sagebionetworks/doi/datacite/DataciteTranslator.java

```
@@ -0,0 +1,78 @@
+	 */
+	public DoiMetadata translate(String xml) {
+		xml.replaceAll("\n","");
+		Document dom = null;
+		try {
+			DocumentBuilder documentBuilder = DocumentBuilderFactoryImpl.newInstance().newDocumentBuilder();
+			dom = documentBuilder.parse(new InputSource(new StringReader(xml)));
+		} catch (ParserConfigurationException e) {
+			throw new RuntimeException("Error occurred while configuring XML parser", e);
+		} catch (SAXException | IOException e) {
+			throw new RuntimeException("Error occurred while parsing metadata", e);
+		}
+
+		// There may be multiple creators or titles
+		DoiMetadata doiMetadata = new DoiMetadata();
+		NodeList xmlCreators = dom.getElementsByTagName("creatorName");
+		List<DoiCreator> metadataCreators = new ArrayList<>();
+		for (int i = 0; i < xmlCreators.getLength(); i++) {
+			DoiCreator creator = new DoiCreator();
+			creator.setCreatorName(xmlCreators.item(i).getTextContent());
+			metadataCreators.add(creator);
+		}
+		doiMetadata.setCreators(metadataCreators);
+
+		NodeList xmlTitles = dom.getElementsByTagName("title");
+		List<DoiTitle> metadataTitles = new ArrayList<>();
+		for (int i = 0; i < xmlTitles.getLength(); i++) {
+			DoiTitle title = new DoiTitle();
+			title.setTitle(xmlTitles.item(i).getTextContent());
+			metadataTitles.add(title);
+		}
+		doiMetadata.setTitles(metadataTitles);
+
+		// There should only be one publication year and resource type, and resource type is not guaranteed to exist
+		NodeList publicationYear = dom.getElementsByTagName("publicationYear");
+		doiMetadata.setPublicationYear(Long.valueOf(publicationYear.item(0).getTextContent()));
+
+		NodeList xmlResourceType = null;
+		DoiResourceType resourceType = new DoiResourceType();
+		try {
+			xmlResourceType = dom.getElementsByTagName("resourceType");
+			resourceType.setResourceTypeGeneral(DoiResourceTypeGeneral.valueOf(xmlResourceType.item(0).getAttributes().getNamedItem("resourceTypeGeneral").getNodeValue()));
+		} catch (NullPointerException e) {
```

Never swallow an exception.  Either throw or log (almost always throw).
Do not use NullPointerException to control follow.